cluster/src/main/scala/org/apache/spark/status/api/v1/MemberDetails.scala

```diff
@@ -0,0 +1,171 @@
+/*
+ * Changes for SnappyData data platform.
+ *
+ * Portions Copyright (c) 2017 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.status.api.v1
+
+import scala.collection.mutable.ListBuffer
+import scala.util.control._
+
+import com.pivotal.gemfirexd.internal.engine.ui.MemberStatistics
+import io.snappydata.SnappyTableStatsProviderService
+
+object MemberDetails {
+
+  def getAllMembersInfo: Seq[MemberSummary] = {
+    val allMembers = SnappyTableStatsProviderService.getService.getMembersStatsFromService
+    val membersBuff: ListBuffer[MemberSummary] = ListBuffer.empty[MemberSummary]
+
+    allMembers.foreach(mem => {
```

Comment:
These types of code can be written in a more functional style like 
allMembers.map( mem => {
getMemberSummary(mem._2)
}
This will return you a sequence.

Reply:
As suggested and discussed using mapValues.